src/emqtt.erl

```diff
@@ -2255,3 +2255,6 @@ redact_packet(#mqtt_packet{variable = #mqtt_packet_connect{} = Conn} = Packet) -
     Packet#mqtt_packet{variable = Conn#mqtt_packet_connect{password = <<"******">>}};
 redact_packet(Packet) ->
     Packet.
+
+flag_to_bool(0) -> false;
```

Comment:
Alternatively, can be fixed in https://github.com/emqx/emqtt/blob/fix-session_present-type/src/emqtt_frame.erl#L218
similarly to how CONNECT packet flags are parsed to bools: https://github.com/emqx/emqtt/blob/fix-session_present-type/src/emqtt_frame.erl#L202

But it would require changing the record field, e.g:
```
#mqtt_packet_connack{session_present  = SessionPresent,
```

Instead of: 
```
#mqtt_packet_connack{ack_flags   = AckFlags,
```